### PR TITLE
Feature/678 operating expenditure

### PR DESCRIPTION
**Pull request by mjtravers**
#678 

@toddlees This PR addresses additional issues pointed out in [PR 861](https://github.com/fecgov/fecfile-web-app/pull/861) plus migrates the redundant Group F transaction types to Group E as discussed with the BAs

### CODE REVIEW

File: front-end/src/app/transactions/transaction-group-b/transaction-group-b.component.ts [hunk 1]
```diff
@@ -28,7 +29,37 @@ export class TransactionGroupBComponent extends TransactionTypeBaseComponent imp
     'memo_code',
     'memo_text_input',
   ];
-  override contactTypeOptions: PrimeOptions = LabelUtils.getPrimeOptions(ContactTypeLabels).filter((option) =>
-    [ContactTypes.INDIVIDUAL, ContactTypes.ORGANIZATION, ContactTypes.COMMITTEE].includes(option.code as ContactTypes)
-  );
+
+  override ngOnInit(): void {
+    if (this.transaction?.transactionType?.scheduleId === 'B') {
```
Comment (toddlees) [inline front-end/src/app/transactions/transaction-group-b/transaction-group-b.component.ts:34]: I see is this what you were talking about pulling out, but held back because of scope creep?

File: front-end/src/app/transactions/transaction-group-e/transaction-group-e.component.html [hunk 2]
```diff
@@ -48,7 +48,7 @@ <h3>Additional Information</h3>
         [descriptionIsSystemGenerated]="isDescriptionSystemGenerated(transaction?.transactionType)"
         [purposeDescriptionLabel]="this.purposeDescriptionLabel"
       ></app-additional-info-input>
-      <div class="grid">
+      <div class="grid" *ngIf="subTransactionOptions.length > 0">
```
Comment (toddlees) [inline front-end/src/app/transactions/transaction-group-e/transaction-group-e.component.html:51]: 💯

File: front-end/src/app/shared/models/transaction-types/PARTNERSHIP_RECOUNT_ACCOUNT_RECEIPT_MEMO.model.ts [hunk 3]
```diff
@@ -2,18 +2,17 @@ import { LabelUtils } from 'app/shared/utils/label.utils';
...
-export class PARTNERSHIP_RECOUNT_ACCOUNT_RECEIPT_MEMO extends SchaTransactionType {
+export class PARTNERSHIP_RECOUNT_ACCOUNT_RECEIPT_MEMO extends SchATransactionType {
   componentGroupId = 'A';
   title = LabelUtils.get(
     ScheduleATransactionTypeLabels,
     ScheduleATransactionTypes.PARTNERSHIP_RECOUNT_ACCOUNT_RECEIPT_MEMO
   );
   schema = schema;
   override updateParentOnSave = true;
-  override subTransactionTypes = [ScheduleATransactionTypes.PARTNERSHIP_RECOUNT_ACCOUNT_RECEIPT_MEMO];
```
Comment (toddlees) [inline front-end/src/app/shared/models/transaction-types/PARTNERSHIP_RECOUNT_ACCOUNT_RECEIPT_MEMO.model.ts:16]: dang nice catch

**Review by toddlees — APPROVED** [review]

File: front-end/src/app/transactions/transaction-group-b/transaction-group-b.component.ts (hunk 1 above)
Comment (mjtravers) [reply front-end/src/app/transactions/transaction-group-b/transaction-group-b.component.ts:34]: I was talking about moving the TransactionTypes and TransactionGroups out of the Transaction file and into the TransactonType file
